pkg/tsdb/azuremonitor/kinds/dataquery/types_dataquery_gen.go

```diff
@@ -0,0 +1,447 @@
+const (
+	UnknownQueryKindUnknownQuery UnknownQueryKind = "UnknownQuery"
+)
+
+// Defines values for WorkspacesQueryKind.
+const (
+	WorkspacesQueryKindWorkspacesQuery WorkspacesQueryKind = "WorkspacesQuery"
+)
+
+// AppInsightsGroupByQueryKind defines model for AppInsightsGroupByQuery.Kind.
+type AppInsightsGroupByQueryKind string
+
+// AppInsightsMetricNameQueryKind defines model for AppInsightsMetricNameQuery.Kind.
+type AppInsightsMetricNameQueryKind string
+
+// Azure Monitor Logs sub-query properties
+type AzureLogsQuery struct {
+	// KQL query to be executed.
+	Query *string `json:"query,omitempty"`
+
+	// @deprecated Use resources instead
+	Resource *string `json:"resource,omitempty"`
+
+	// Array of resource URIs to be queried.
+	Resources *[]string `json:"resources,omitempty"`
+
+	// Specifies the format results should be returned as.
+	ResultFormat *AzureLogsQueryResultFormat `json:"resultFormat,omitempty"`
+
+	// Workspace ID. This was removed in Grafana 8, but remains for backwards compat
+	Workspace *string `json:"workspace,omitempty"`
+}
+
+// Specifies the format results should be returned as.
+type AzureLogsQueryResultFormat string
+
+// AzureMetricDimension defines model for AzureMetricDimension.
+type AzureMetricDimension struct {
+	// Name of Dimension to be filtered on.
+	Dimension *string `json:"dimension,omitempty"`
+
+	// @deprecated filter is deprecated in favour of filters to support multiselect.
+	Filter *string `json:"filter,omitempty"`
+
+	// Values to match with the filter.
+	Filters *[]string `json:"filters,omitempty"`
+
+	// String denoting the filter operation. Supports 'eq' - equals,'ne' - not equals, 'sw' - starts with. Note that some dimensions may not support all operators.
+	Operator *string `json:"operator,omitempty"`
+}
+
+// AzureMetricQuery defines model for AzureMetricQuery.
+type AzureMetricQuery struct {
+	// The aggregation to be used within the query. Defaults to the primaryAggregationType defined by the metric.
+	Aggregation *string `json:"aggregation,omitempty"`
+
+	// Aliases can be set to modify the legend labels. e.g. {{ resourceGroup }}. See docs for more detail.
+	Alias *string `json:"alias,omitempty"`
+
+	// Time grains that are supported by the metric.
+	AllowedTimeGrainsMs *[]int64 `json:"allowedTimeGrainsMs,omitempty"`
+
+	// Used as the value for the metricNamespace property when it's different from the resource namespace.
+	CustomNamespace *string `json:"customNamespace,omitempty"`
+
+	// @deprecated This property was migrated to dimensionFilters and should only be accessed in the migration
+	Dimension *string `json:"dimension,omitempty"`
+
+	// @deprecated This property was migrated to dimensionFilters and should only be accessed in the migration
+	DimensionFilter *string `json:"dimensionFilter,omitempty"`
+
+	// Filters to reduce the set of data returned. Dimensions that can be filtered on are defined by the metric.
+	DimensionFilters *[]struct {
```

Comment:
@sdboyer this is what I meant (sorry if my explanation was a bit confusing). This `DimensionFilters` struct is defined above on line 155 as `AzureMetricDimension` so I would expect it to be referenced here but instead the whole `AzureMetricDimension` struct is duplicated.

Reply:
Sorry, yes, this is the flipside of the thing i was describing, but i put it poorly.

These should clearly just be a reference to the other generated type. But they're getting _incorrectly inlined_ right now because of the config option we have to set in order to _correctly inline_ `DataQuery` itself.

It's a bug, and we'll fix it 😄 

(also ughhh we've got slice pointers again! @spinillos )